include/swift/AST/TypeCheckRequests.h

```diff
@@ -2311,6 +2311,25 @@ class SpecializeAttrTargetDeclRequest
   bool isCached() const { return true; }
 };
 
+/// Looks up and applies the access note for a given declaration.
+class ApplyAccessNoteRequest
+    : public SimpleRequest<ApplyAccessNoteRequest,
+                           evaluator::SideEffect(ValueDecl *),
```

Comment:
Do you need `SideEffect` for a single return value?

Reply:
Ah, because the result value is dropped at the call sites I guess? I understand the use of `SideEffect` in this case.

Reply:
Yeah. `ApplyAccessNoteRequest` mutates the AST; I'm basically using the request evaluator here to avoid applying the access notes twice or reentering access note application for the same declaration.